capstone/fitness/src/main/java/com/google/sps/progress/Progress.java

```diff
@@ -0,0 +1,220 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.progress;
+
+import com.google.sps.fit.*;
+import com.google.sps.fit.Exercise.SetType;
+import com.google.sps.util.*;
+import java.util.*;
+
+/* Handler for ProgressModel updates. */
+public class Progress {
```

Comment:
I feel there's no need for this class. All the functions it performs could be included in ProgressModel. You can init ProgressModel with Data and compute the updated GoalStep.

Reply:
You're right, I wanted something else to shape how the model was built, but thinking about it now, it doesn't really make sense. It has been changed.